frontend/src/app/App.test.tsx

```diff
@@ -230,311 +237,378 @@ describe("App", () => {
-            scriptStopRequested: true,
-          }),
-        }),
+    it("sends theme info to the host when the app is first rendered", () => {
+      instance.componentDidMount()
+
+      expect(sendMessageFunc).toHaveBeenCalledWith({
+        type: "SET_THEME_CONFIG",
+        themeInfo: toExportedTheme(lightTheme.emotion),
       })
-    )
+    })
 
-    expect(wrapper.state("scriptRunState")).toBe(ScriptRunState.STOP_REQUESTED)
-    expect(instance.props.hostCommunication.onScriptStop).toHaveBeenCalled()
-  })
+    it("closes modals when the modal closure message has been received", () => {
+      const dialog = StreamlitDialog({
+        type: DialogType.ABOUT,
+        sessionInfo: mockSessionInfo(),
+        onClose: () => {},
+      })
 
-  it("changes scriptRunState and fires withHostCommunication callback when scriptRerunRequested signal has been received", () => {
-    const wrapper = shallow(<App {...getProps()} />)
-    const instance = wrapper.instance() as App
+      wrapper.setState({ dialog })
+      expect(wrapper.find(Modal)).toHaveLength(1)
 
-    instance.isServerConnected = jest.fn().mockReturnValue(true)
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "CLOSE_MODAL",
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
+
+      expect(wrapper.find(Modal)).toHaveLength(0)
+    })
+
+    it("changes scriptRunState and triggers stopScript when STOP_SCRIPT message has been received", () => {
+      instance.isServerConnected = jest.fn().mockReturnValue(true)
 
-    wrapper.setProps(
-      getProps({
-        hostCommunication: getHostCommunicationProp({
-          currentState: getHostCommunicationState({
-            scriptRerunRequested: true,
-          }),
-        }),
+      // We explicitly set the scriptRunState to RUNNING, so we can test that
+      // scriptStopRequested is handled correctly.
+      wrapper.setState({
+        scriptRunState: ScriptRunState.RUNNING,
       })
-    )
 
-    expect(wrapper.state("scriptRunState")).toBe(
-      ScriptRunState.RERUN_REQUESTED
-    )
-    expect(instance.props.hostCommunication.onScriptRerun).toHaveBeenCalled()
-  })
+      const stopScriptFunc = jest.spyOn(
+        instance.hostCommunicationMgr.props,
+        "stopScript"
+      )
 
-  it("fires withHostCommunication callback when cacheClearRequested signal has been received", () => {
-    const wrapper = shallow(<App {...getProps()} />)
-    const instance = wrapper.instance() as App
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "STOP_SCRIPT",
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
 
-    instance.isServerConnected = jest.fn().mockReturnValue(true)
+      expect(stopScriptFunc).toHaveBeenCalled()
```

Comment:
Let's use `toHaveHaveBeenCalledWith()` here and every other modified test in this file, for more precision